jwst/tweakreg/tweakreg_step.py

```diff
@@ -200,6 +207,61 @@ def process(self, input):
             else:
                 raise e
 
+        if self.align_to_gaia:
+            try:
+                # Get catalog of GAIA sources for the field
+                #
+                # NOTE:  If desired, the pipeline can write out the reference catalog
+                #        as a separate product with a name based on whatever convention
+                #        is determined by the JWST Cal Working Group
+                if self.output_gaia:
+                    output_name = 'fit_{}_ref.ecsv'.format(self.gaia_catalog.lower())
+                else:
+                    output_name = None
+                ref_cat = amutils.create_astrometric_catalog(self.input_models,
+                                                             self.gaia_catalog,
+                                                             output=output_name)
+
+                # Check that there are enough GAIA sources for a reliable/valid fit
+                num_ref = len(ref_cat)
+                if num_ref < self.minobj:
+                    msg = "Not enough GAIA sources for a fit: {}\n".format(num_ref)
+                    msg += "Skipping alignment to {} astrometric catalog!\n".format(self.gaia_catalog)
+                    raise ValueError(msg)
+                # Set group_id to same value so all get fit as one observation
+                for imcat in imcats:
+                    imcat.meta['orig_group_id'] = imcat.meta['group_id']
+                    imcat.meta['group_id'] = 987654
```

Comment:
Any particular significance to that number or just picked it at random?

Reply:
This number is a random large number unlikely to be used in any program for the group_id identifier within any association.  The distinct value also makes it clear that it is an artificially assigned ID which can be recognized as being associated with alignment to GAIA as opposed to the group_id's used for relative alignment of the same observations performed earlier in this step.

Reply:
`imcat.meta['group_id']` is used exclusively internally for alignment purpose and it is not stored anywhere in the `DataModel`. It can be anything. Setting it to 1 would be just as good as setting it to 987654.